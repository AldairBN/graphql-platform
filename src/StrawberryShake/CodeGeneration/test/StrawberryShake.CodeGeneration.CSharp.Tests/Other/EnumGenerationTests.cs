using System.Text;
using System.Threading.Tasks;
using Snapshooter.Xunit;
using StrawberryShake.CodeGeneration;
using StrawberryShake.CodeGeneration.CSharp;
using Xunit;

namespace StrawberryShake.Other
{
    public class EnumGenerationTests
    {
        readonly StringBuilder _stringBuilder;
        readonly CodeWriter _codeWriter;
        readonly EnumGenerator _generator;

        public EnumGenerationTests()
        {
            _stringBuilder = new StringBuilder();
            _codeWriter = new CodeWriter(_stringBuilder);
            _generator = new EnumGenerator();
        }

        [Fact]
        public async Task GenerateSingleValueEnum()
        {
            await _generator.Write(
                _codeWriter,
                new EnumDescriptor(
                    "Foonum",
                    "FooBarNamespace",
                    new []
                    {
                        new EnumElementDescriptor("Foo", "foo")
                    }
                )
            );
            _stringBuilder.ToString().MatchSnapshot();
        }

        [Fact]
        public async Task GenerateSingleValueEnumWithExplicitValue()
        {
            await _generator.Write(
                _codeWriter,
                new EnumDescriptor(
                    "Foonum",
                    "FooBarNamespace",
                    new []
                    {
                        new EnumElementDescriptor("Foo", "foo", 42)
                    }
                )
            );
            _stringBuilder.ToString().MatchSnapshot();
        }

        [Fact]
        public async Task GenerateMultiValueEnum()
        {
            await _generator.Write(
                _codeWriter,
                new EnumDescriptor(
                    "Foonum",
                    "FooBarNamespace",
                    new []
                    {
                        new EnumElementDescriptor("Foo", "foo"),
                        new EnumElementDescriptor("Bar", "bar")
                    }
                )
            );
            _stringBuilder.ToString().MatchSnapshot();
        }

        [Fact]
        public async Task GenerateMultiValueEnumWithOneExplicitValue()
        {
            await _generator.Write(
                _codeWriter,
                new EnumDescriptor(
                    "Foonum",
                    "FooBarNamespace",
                    new []
                    {
                        new EnumElementDescriptor("Foo", "foo", 42),
                        new EnumElementDescriptor("Bar", "bar")
                    }
                )
            );
            _stringBuilder.ToString().MatchSnapshot();
        }

        [Fact]
        public async Task GenerateMultiValueEnumWithMultipleExplicitValues()
        {
            await _generator.Write(
                _codeWriter,
                new EnumDescriptor(
                    "Foonum",
                    "FooBarNamespace",
                    new []
                    {
                        new EnumElementDescriptor("Foo", "foo", 21),
                        new EnumElementDescriptor("Bar", "bar", 42)
                    }
                )
            );
            _stringBuilder.ToString().MatchSnapshot();
        }
    }
}
