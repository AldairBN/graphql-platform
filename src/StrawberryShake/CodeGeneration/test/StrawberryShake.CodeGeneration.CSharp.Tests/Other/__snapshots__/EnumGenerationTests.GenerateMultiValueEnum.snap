namespace FooBarNamespace
{
    [global::System.CodeDom.Compiler.GeneratedCode("StrawberryShake", "11.0.0")]
    public enum Foonum
    {
        Foo,
        Bar
    }
}
